Sources/CvPlayer.cpp

```diff
@@ -19146,12 +19100,9 @@ void CvPlayer::updateGroupCycle(CvUnit* pUnit, bool bFarMove)
 						"CvGameCoreDLL Diagnostics",
 						MB_OK);
 	}
-	pUnitNode = pPlot->headUnitNode();
 
-	while (pUnitNode != NULL)
+	foreach_ (CvUnit* pLoopUnit, pUnit->plot()->units())
 	{
-		pLoopUnit = ::getUnit(pUnitNode->m_data);
-		pUnitNode = pPlot->nextUnitNode(pUnitNode);
 		//AlbertS2 added this to fix a CTD that happened while loading a saved game
```

Comment:
We should remove things like this, code shouldn't be changed to fix broken saved games like this. It should be constrained to either the loading code or the modifier recalc.